### PR TITLE
Decimal: FormulaTypeJsonConverter

### PR DESCRIPTION
**Pull request by nguhoa**
Fix issue: https://github.com/microsoft/Power-Fx/issues/1316

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs [hunk 1]
```diff
@@ -1117,7 +1118,7 @@ public void TestPublishTokens(string text, int count, bool withAllowSideEffects,
         [InlineData("{\"A\": 1 }", "A+2", typeof(DecimalType))]
         [InlineData("{}", "\"hi\"", typeof(StringType))]
         [InlineData("{}", "", typeof(BlankType))]
-        [InlineData("{}", "{ A: 1 }", typeof(KnownRecordType))]
+        [InlineData("{}", "{ A: 1 }", typeof(UserDefinedRecordType))]
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs:1121]: >UserDefinedRecordType [](http://example.com/codeflow?start=45&length=21)

Confirming this change is intended?

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs [hunk 2]
```diff
@@ -1177,12 +1178,12 @@ public void TestPublishExpressionType_Null(string context, string expression)
         }
 
         [Theory]
-        [InlineData(false, "{}", "{ A: 1 }", @"{""Type"":""Record"",""Fields"":{""A"":{""Type"":""Decimal""}}}")]
-        [InlineData(false, "{}", "[1, 2]", @"{""Type"":""Table"",""Fields"":{""Value"":{""Type"":""Decimal""}}}")]
-        [InlineData(true, "{}", "[{ A: 1 }, { B: true }]", @"{""Type"":""Table"",""Fields"":{""A"":{""Type"":""Decimal""},""B"":{""Type"":""Boolean""}}}")]
-        [InlineData(false, "{}", "[{ A: 1 }, { B: true }]", @"{""Type"":""Table"",""Fields"":{""Value"":{""Type"":""Record"",""Fields"":{""A"":{""Type"":""Decimal""},""B"":{""Type"":""Boolean""}}}}}")]
-        [InlineData(false, "{}", "{A: 1, B: { C: { D: \"Qwerty\" }, E: true } }", @"{""Type"":""Record"",""Fields"":{""A"":{""Type"":""Decimal""},""B"":{""Type"":""Record"",""Fields"":{""C"":{""Type"":""Record"",""Fields"":{""D"":{""Type"":""String""}}},""E"":{""Type"":""Boolean""}}}}}")]
-        [InlineData(false, "{}", "{ type: 123 }", @"{""Type"":""Record"",""Fields"":{""type"":{""Type"":""Decimal""}}}")]
+        [InlineData(false, "{}", "{ A: 1 }", @"{""type"":{""name"":""Record""},""fields"":{""A"":{""type"":{""name"":""Decimal""}}}}")]
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs:1181]: >@"{"" [](http://example.com/codeflow?start=45&length=5)

Did the json format here change?

Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs:1181]: yes, json format here is different.

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs (hunk 1 above)
Comment (nguhoa) [reply src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/LanguageServerTests.cs:1121]: with this new change, it returned that type. not the knownrecordtype.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by nguhoa** [comment]
https://github.com/microsoft/Power-Fx/issues/1316 is fixed so closed this PR

**Pull request closed without merging**